src/shims/io.rs

```diff
@@ -262,7 +264,10 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         match result {
             Ok(ok) => Ok(ok),
             Err(e) => {
-                self.eval_context_mut().machine.last_error = e.raw_os_error().unwrap() as u32;
+                self.eval_context_mut().set_last_error(Scalar::from_int(
```

Comment:
This pattern appears a few times, maybe create a custom method for `io::Error`?

Reply:
Done